### PR TITLE
Experimental sim with concurrent links

### PR DESCRIPTION
**Pull request by michael-isaev**

### CODE REVIEW

File: paragraph/scheduling/graph_scheduler.cc [hunk 1]
```diff
@@ -83,7 +83,7 @@ void GraphScheduler::SetLogger(std::unique_ptr<Logger> logger) {
 void GraphScheduler::InstructionStarted(
     Instruction* instruction, double current_time) {
   CHECK(initialized_);
-  CHECK_LE(current_time_, current_time);
+  // CHECK_LE(current_time_, current_time);
```
Comment (nicmcd) [inline paragraph/scheduling/graph_scheduler.cc]: Leaving dead code? Why are you removing this?

File: paragraph/simulator/simple_sim.cc [hunk 2]
```diff
@@ -100,7 +132,7 @@ absl::Status SimpleSim::FetchAndExecute() {
 }
 
 SimpleSim::SimpleSim(const PerformanceParameters& processor_parameters)
```
Comment (nicmcd) [inline paragraph/simulator/simple_sim.cc]: Why don't these names match?

File: paragraph/simulator/simple_sim.cc [hunk 3]
```diff
@@ -69,22 +72,51 @@ absl::Status SimpleSim::FetchAndExecute() {
...
+      RETURN_IF_FALSE(rx_link_time_.find(src) != rx_link_time_.end(),
+                      absl::InternalError) << "PeerId " << src
+          << " not found in simulator receiving links map.";
+      double current_time = std::max(
+          scheduler_->GetFsm(executing_instruction).GetTimeReady(),
+          rx_link_time_.at(src));
+      scheduler_->InstructionStarted(executing_instruction, current_time);
+      if (opcode == Opcode::kRecvDone) {
+        // We take into account network delay on rx_link between current
+        // processor and its peer for RecvDone instructions, while we modes
```
Comment (nicmcd) [inline paragraph/simulator/simple_sim.cc]: "while we modes"???

File: paragraph/simulator/simple_sim.cc [hunk 4]
```diff
@@ -37,8 +37,12 @@ shim::StatusOr<std::unique_ptr<SimpleSim>> SimpleSim::Create(
 }
 
 absl::Status SimpleSim::StartSimulation(double start_time) {
```
Comment (nicmcd) [inline paragraph/simulator/simple_sim.cc]: This should really be called Simulate(). StartSimulation() leads you to believe it is a non-blocking function call.

File: paragraph/scheduling/graph_scheduler.cc (hunk 1 above)
Comment (michael-isaev) [reply paragraph/scheduling/graph_scheduler.cc]: updated with comment

File: paragraph/simulator/simple_sim.cc (hunk 2 above)
Comment (michael-isaev) [reply paragraph/simulator/simple_sim.cc]: fixed

File: paragraph/simulator/simple_sim.cc (hunk 3 above)
Comment (michael-isaev) [reply paragraph/simulator/simple_sim.cc]: fixed

File: paragraph/simulator/simple_sim.cc (hunk 4 above)
Comment (michael-isaev) [reply paragraph/simulator/simple_sim.cc]: updated

**Review by nicmcd — APPROVED** [review]

**Pull request merged**